examples/manipulation-demo-streamlit.py

```diff
@@ -0,0 +1,112 @@
+# Copyright (C) 2024 Robotec.AI
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language goveself.rning permissions and
+# limitations under the License.
+
+import rclpy
+import rclpy.qos
+import streamlit as st
+from langchain_core.messages import AIMessage, HumanMessage, ToolMessage
+
+from rai.agents.conversational_agent import create_conversational_agent
+from rai.agents.integrations.streamlit import get_streamlit_cb, streamlit_invoke
+from rai.messages import HumanMultimodalMessage
+from rai.node import RaiBaseNode
+from rai.tools.ros.manipulation import GetObjectPositionsTool, MoveToPointTool
+from rai.tools.ros.native import GetCameraImage, Ros2GetTopicsNamesAndTypesTool
+from rai.utils.model_initialization import get_llm_model
+
+
+@st.cache_resource
+def initialize_graph():
+    rclpy.init()
+    node = RaiBaseNode(node_name="manipulation_demo")
+    node.declare_parameter("conversion_ratio", 1.0)
+    node.qos_profile.reliability = rclpy.qos.ReliabilityPolicy.RELIABLE
+
+    tools = [
+        GetObjectPositionsTool(
+            node=node,
+            target_frame="panda_link0",
+            source_frame="RGBDCamera5",
+            camera_topic="/color_image5",
+            depth_topic="/depth_image5",
+            camera_info_topic="/color_camera_info5",
+        ),
+        MoveToPointTool(node=node, manipulator_frame="panda_link0"),
+        GetCameraImage(node=node),
+        Ros2GetTopicsNamesAndTypesTool(node=node),
+    ]
+
+    system_prompt = """
+    You are a robotic arm with interfaces to detect and manipulate objects.
+    Here are the coordinates information:
+    x - front to back (positive is forward)
+    y - left to right (positive is right)
+    z - up to down (positive is up)
+
+    Before starting the task, make sure to grab the camera image to understand the environment.
+    """
+
+    llm = get_llm_model(model_type="complex_model", streaming=True)
+    agent = create_conversational_agent(
+        llm=llm,
+        tools=tools,
+        system_prompt=system_prompt,
+    )
+    return agent
```

Comment:
This part is copied from [here](https://github.com/RobotecAI/rai/blob/47e74fe34d141274783ceb8a535c420c2bcc6758/examples/manipulation-demo.py#L24-L58). Please refactor

Reply:
Done.

Reply:
for some reason after the change (or rebase) streamlit doesn't work for me:
![image](https://github.com/user-attachments/assets/35778fb9-d6fe-47b1-9782-3796e7271f12)

Reply:
I tried `streamlit run src/rai/rai/tools/debugging_assistant.py` and it works well. So I think it's sth wrong with this PR

Reply:
I can confirm it's not working correctly. The agent does call tools and respond, but the last message seems to only be displayed after the user types in another message. I'm looking into this now.

Reply:
The issue should be solved now. I made a small mistake when refactoring the duplicated code and missed it when testing.

Reply:
now it works great! thanks